src/SymbolFinder.php

```diff
@@ -35,31 +35,56 @@ class SymbolFinder extends NodeVisitorAbstract
      */
     private $containerName;
 
+    /**
+     * @var array
+     */
+    private $nameStack = array();
+
+    /**
+     * @var array
+     */
+    private $nodeStack = array();
+
+    /**
+     * @var int
+     */
+    private $functionCount = 0;
+
     public function __construct(string $uri)
     {
         $this->uri = $uri;
     }
 
     public function enterNode(Node $node)
     {
+        array_push($this->nodeStack, $node);
+        $containerName = end($this->nameStack);
+
+        // If we enter a named node, push its name onto name stack.
+        // Else push the current name onto stack.
+        if (isset($node->name) && is_string($node->name) && !empty($node->name)){
+            array_push($this->nameStack, $node->name);
+        }
+        else {
+            array_push($this->nameStack, $containerName);
+        }
+
         $class = get_class($node);
         if (!isset(self::NODE_SYMBOL_KIND_MAP[$class])) {
             return;
         }
 
+        // if we enter a method or function, increase the function counter
+        if ($class === Node\Stmt\Function_::class || $class === Node\Stmt\ClassMethod::class) {
```

Comment:
No particular reason, just reused the $class variable. Instanceof would probably make the purpose clearer.